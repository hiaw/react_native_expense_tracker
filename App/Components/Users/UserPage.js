import React, { Component } from 'react'
import { Alert, View, Button } from 'react-native'
import { observable } from 'mobx'
import { observer } from 'mobx-react'
import Moment from 'moment'
import { FormLabel, FormInput } from 'react-native-elements'
import { Actions } from 'react-native-router-flux'

import styles from './UsersList.style.js'

@observer
export default class UserView extends Component {
  @observable editing = false
  @observable email = ''
  @observable password = ''
  @observable role = ''

  constructor (props) {
    super(props)

    this.userService = props.app.service('users')

    if (props.user) {
      this.editing = true
      this.email = props.user.email
      this.role = props.user.role
    }
  }

  add () {
    this.userService.create({
      email: this.email,
      password: this.password,
      role: this.role
    }).then((res) => {
      Actions.pop()
    })
  }

  save () {
    this.userService.update(this.props.user._id, {
      email: this.amount,
      role: this.role
    }).then((res) => {
      Actions.pop()
    })
  }

  delete () {
    Alert.alert(
      'Delete this user',
      'Are you sure?',
      [
        {text: 'Yes', onPress: () => {
          this.userService.remove(this.props.user._id)
            .then((res) => {
              Actions.pop()
            })
        }},
        {text: 'Cancel', onPress: () => {}, style: 'cancel'}
      ]
    )
  }

  render () {
    let dateText = Moment(this.date).format('DD/MM/YY')
    let timeText = Moment(this.date).format('HH:mm')

    return (
      <View style={styles.container}>
        <FormLabel>Role</FormLabel>
        <FormInput value={this.role}
          onChangeText={(t) => { this.role = t }} />

        <FormLabel>Email</FormLabel>
        <FormInput value={this.email}
          onChangeText={(t) => { this.email = t }} />

        { this.editing ? this.renderEditButtons() : this.renderAddButton()}

      </View>
    )
  }

  renderAddButton () {
    return (
      <View>
        <FormLabel>Password</FormLabel>
        <FormInput value={this.password}
          secureTextEntry
          onChangeText={(t) => { this.password = t }} />

        <Button title='Add'
          onPress={() => this.add()} />
      </View>
    )
  }

  renderEditButtons () {
    return (
      <View>
        <Button title='Save'
          onPress={() => this.save()} />
        <Button title='Delete'
          color='red'
          onPress={() => this.delete()} />
      </View>
    )
  }
}
