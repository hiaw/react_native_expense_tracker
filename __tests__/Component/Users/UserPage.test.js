import 'react-native'
import React from 'react'
import UserPage from '../../../App/Components/Users/UserPage.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

let user = {
  _id: 'adsflkja',
  email: 'test@test.com',
  roles: ['admin']
}

let app = { service: () => {}}

it('renders correctly', () => {
  const tree = renderer.create(
    <UserPage user={user} app={app} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
