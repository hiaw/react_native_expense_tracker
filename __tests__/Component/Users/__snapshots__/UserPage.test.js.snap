// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 65,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Role
    </Text>
  </View>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Admin",
            "textColor": undefined,
            "value": "admin",
          },
          Object {
            "label": "Manager",
            "textColor": undefined,
            "value": "manager",
          },
          Object {
            "label": "User",
            "textColor": undefined,
            "value": "user",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": undefined,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Email
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
        },
        undefined,
      ]
    }
  >
    <TextInput
      autoCapitalize={undefined}
      autoCorrect={undefined}
      autoFocus={undefined}
      blurOnSubmit={undefined}
      clearButtonMode={undefined}
      clearTextOnFocus={undefined}
      dataDetectorTypes={undefined}
      defaultValue={undefined}
      editable={undefined}
      enablesReturnKeyAutomatically={undefined}
      inlineImageLeft={undefined}
      inlineImagePadding={undefined}
      keyboardAppearance={undefined}
      keyboardType={undefined}
      maxLength={undefined}
      multiline={undefined}
      numberOfLines={undefined}
      onBlur={undefined}
      onChange={undefined}
      onChangeText={[Function]}
      onContentSizeChange={undefined}
      onEndEditing={undefined}
      onFocus={undefined}
      onKeyPress={undefined}
      onLayout={undefined}
      onSelectionChange={undefined}
      onSubmitEditing={undefined}
      placeholder={undefined}
      placeholderTextColor={undefined}
      returnKeyLabel={undefined}
      returnKeyType={undefined}
      secureTextEntry={undefined}
      selectTextOnFocus={undefined}
      selectionColor="#86939e"
      selectionState={undefined}
      style={
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 36,
          },
          undefined,
        ]
      }
      underlineColorAndroid={undefined}
      value="test@test.com"
    />
  </View>
  <View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          Delete
        </Text>
      </View>
    </View>
  </View>
</View>
`;
